Make remote commands smarter

mwleeds opened the pull request:
See the commit messages for the details. This is a fix for #787

alexlarsson commented on app/flatpak-builtins-list-remotes.c:
```diff
@@ -194,10 +150,10 @@ gboolean
   context = g_option_context_new ("");
-  if (!flatpak_option_context_parse (context, options, &completion->argc, &completion->argv, 0, &dir, NULL, NULL))
+  if (!flatpak_option_context_parse (context, options, &completion->argc, &completion->argv,
+                                     FLATPAK_BUILTIN_FLAG_ONE_DIR, NULL, NULL, NULL))
```
Flags should be 0 here, no?

mwleeds replied:
yeah, not sure what I was thinking

mwleeds replied:
Now that I'm looking at it again I remember; it was that I was trying to maintain that the first commit didn't introduce any functional changes. So I'll move that change into a separate commit.

alexlarsson commented on app/flatpak-builtins-utils.c:
```diff
@@ -326,3 +326,59 @@ flatpak_load_gpg_keys (char **gpg_import,
+  if (out_dir)
+    {
+      if (dirs_with_remote->len == 0)
+        *out_dir = NULL;
```
Every single user of this handles it by picking a random dir and letting some other operation fail. I think it makes a lot more sense to have this report an error here.

alexlarsson commented on app/flatpak-builtins-ls-remote.c:
```diff
@@ -50,12 +51,33 @@ static GOptionEntry options[] = {
 
+typedef struct RemoteDirPair {
+  gchar *remote_name;
+  gchar *dir_id;
```
I don't think using the dir_id for this works. For instance, the user directory has a NULL directory id, but when you later call flatpak_dir_get_system_by_id() with this you'll get back the system dir. Why not have a ref to the actual directory

mwleeds commented:
Alright @alexlarsson I addressed your comments, and I changed `flatpak_resolve_duplicate_remotes` so it doesn't segfault if a choice has to be made and it's run from a script.

alexlarsson commented:
@rh-atomic-bot r+

alexlarsson commented:
bot, retest this please

alexlarsson commented:
```
# error: Remote "test-repo" not found
+ echo 'File '\''remote-modify-error-log'\'' doesn'\''t match regexp '\''No remote test-repo'\'''
```
I think you changed some string...

mwleeds commented:
bot, retest this please

alexlarsson commented:
@rh-atomic-bot r+
